pkg/registry/common/querycache/nse_client.go

```diff
@@ -82,9 +100,27 @@ func (q *queryCacheNSEClient) Find(ctx context.Context, in *registry.NetworkServ
 		}()
 	}
 	close(resultCh)
+
 	return streamchannel.NewNetworkServiceEndpointFindClient(ctx, resultCh), nil
 }
 
+func (q *queryCacheNSEClient) checkQuery(ctx context.Context, nseQuery *registry.NetworkServiceEndpointQuery, opts ...grpc.CallOption) error {
+	nseQuery = proto.Clone(nseQuery).(*registry.NetworkServiceEndpointQuery)
+	nseQuery.Watch = false
+
+	findCtx, findCancel := context.WithCancel(q.chainCtx)
+	defer findCancel()
+
+	client, err := next.NetworkServiceEndpointRegistryClient(ctx).Find(findCtx, nseQuery, opts...)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Recv()
+
+	return err
+}
```

Comment:
Could you explain why do we need this?

Reply:
I have removed `checkQuery`, but here is an issue: #678.